client/src/endpoint/account.rs

```diff
@@ -452,3 +452,180 @@ mod effects_tests {
         assert_eq!(req.uri().query(), Some("cursor=CURSOR&order=asc&limit=123"));
     }
 }
+
+/// An endpoint that accesses the Payment operations where a specific account is either the
+/// sender or receiver.
+#[derive(Debug, Clone)]
+pub struct Payments {
+    id: String,
+    cursor: Option<String>,
+    order: Option<Order>,
+    limit: Option<u32>,
+}
+
+impl Payments {
+    /// Returns a new endpoint for account payments. Hand this to the client in order to request
+    /// payment operations for a specific account.
+    pub fn new(id: &str) -> Self {
+        Self {
+            id: id.to_string(),
+            cursor: None,
+            order: None,
+            limit: None,
+        }
+    }
+
+    /// Starts the page of results at a given cursor
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use stellar_client::sync::Client;
+    /// use stellar_client::endpoint::account;
+    /// # use stellar_client::endpoint::transaction;
+    /// # use stellar_client::endpoint::Order;
+    ///
+    /// let client   = Client::horizon_test().unwrap();
+    /// # let transaction_ep   = transaction::All::default().limit(1);
+    /// # let txns             = client.request(transaction_ep).unwrap();
+    /// # let txn              = &txns.records()[0];
+    /// # let account_id       = txn.source_account();
+    /// # let endpoint         = account::Transactions::new(account_id);
+    /// # let first_page       = client.request(endpoint).unwrap();
+    /// # // grab first page and extract cursor
+    /// # let cursor           = first_page.next_cursor();
+    /// let endpoint = account::Payments::new(account_id).cursor(cursor);
+    /// let records  = client.request(endpoint).unwrap();
```

Comment:
Some of these endpoints require a lot of work to test :)  In this case it looks like the cursor is coming from the transactions endpoint and that we are using it to find our place in the payments endpoint.  I'll admit that I'm not sure if the cursor is a global counter that increments all resources and the docs aren't terribly clear on this either https://www.stellar.org/developers/horizon/reference/resources/page.html

Assuming we did get the cursor from payments, there is still no guarantee that the first account returned from the transaction call has 2 payments, so this test may still blink anyhow which is a bummer.  We have been moving towards testing that the URL is properly constructed and testing the actual cursor functionality on the `All` endpoint.  https://github.com/kbacha/stellar-sdk/blob/master/client/src/endpoint/ledger.rs#L624-L626  I think this should be fine here.

Sorry for the long stream of consciousness response :)

Reply:
I believe that @correlator is correct, transactions are not guaranteed to have payments (account creation and offers are in transactions but no payment is made i believe). 

You may try using the `payment::All` endpoint to make sure that you locate an actual payment, that would give you guarantees.